resources/bucket_list.py

```diff
@@ -0,0 +1,245 @@
+        bucket_list = BucketList.query.filter_by(id=bucketlist_id).first()
+
+        if bucket_list:
+            bucket_list_item_last = BucketListItem.query.filter_by(bucketlist_id=bucketlist_id)\
+                                   .order_by(db.desc(BucketListItem.bucketlist_item_id))
+
+            try:
+                last_item_id = bucket_list_item_last[0].bucketlist_item_id
+            except IndexError:
+                last_item_id = 0
+
+            try:
+                bucket_list_item = BucketListItem(bucketlist_id=bucketlist_id, name=name,\
+                    bucketlist_item_id=last_item_id + 1)
+                db.session.add(bucket_list_item)
+                db.session.commit()
+                return marshal(bucket_list_item, bucket_list_item_field), 201
+            except:
+                return {'message':'item already exists'}, 202
+
+        return {'error':'bucket list cannot be found'}, 404
+
+    def put(self, bucketlist_id, bucketlist_item_id):
+        """
+        updates a particular bucketlist item of  a particular id
+        """
+
+        self.reqparse.add_argument('done', type=bool,
+                                   help='No done argument provided',
+                                   location='json')
+        self.reqparse.add_argument('item_name', type=str,
+                                   help='No item name provided',
+                                   location='json')
+        args = self.reqparse.parse_args()
+        name = args['item_name']
+        done = args['done']
+
+        bucket_list = BucketList.query.filter_by(id=bucketlist_id, created_by=g.user.id).first()
+
+        if bucket_list:
+            bucket_list_item = BucketListItem.query.\
+                                     filter_by(bucketlist_item_id=bucketlist_item_id).first()
+            if bucket_list_item:
+                if name:
+                    bucket_list_item.name = name
+                if done:
+                    bucket_list_item.done = True
+                db.session.commit()
+                return marshal(bucket_list_item, bucket_list_item_field), 200
+
+            return {'error':'bucket list item does not exist'}, 202
+
+        return {'error':'bucket list does not exist'}, 202
+
+    def delete(self, bucketlist_id, bucketlist_item_id):
+        """ deletes a particular bucket list item"""
+
+        bucket_list = BucketList.query.filter_by(id=bucketlist_id,\
+                                                created_by=g.user.id).first()
+
+        if bucket_list:
+            bucket_list_item = BucketListItem.query.filter_by(id=bucketlist_item_id).first()
+
+            if bucket_list_item:
+                db.session.delete(bucket_list_item)
+                db.session.commit()
+                return {'message':'bucket list item deleted'}, 200
+
+            return {'message':'bucket list item does not exist'}, 404
+
+        return {'message':'bucket list does not exist'}, 404
+
+
```

Comment:
Same case as the first statement. Consider using `if not` then a return statement after.